### ISSUE TITLE
[BUG] - 

### ISSUE THREAD

**Issue by cardano-apexpool** [input-output-hk/cardano-node#3659]
**Internal/External**
*External*  

**Area**
*Other*   Submitting transaction with cardano-submit-api

**Summary**
A transaction created with cardano-cli 1.32.1 on another computer can be send to a cardano-submit-api 1.33.0 running on a server with cardano-node 1.33.0, but when sending an identical transaction to the same server after upgrading cardano-submit-api and cardano-node to 1.34.0-rc3 (git revision 54155a28dbdb9a17e4ffe52b68f6c58af54ff83b), sending the transaction fails.
The message in the cardano-submit-api log is (I tried 2 times, and a few times yesterday):

    Feb 28 16:10:04 <hostname> bash[92594]: MuxError (MuxIOException writev: resource vanished (Broken pipe)) "(sendAll errored)"
    Feb 28 16:12:16 <hostname> bash[92594]: MuxError (MuxIOException writev: resource vanished (Broken pipe)) "(sendAll errored)"

After downgrading the cardano-node and the cardano-submit-api to 1.33.0, submitting the same transaction file is working as expected.

**Steps to reproduce**
1. Create the transaction:

        ADDRESS=$(cat wallet/dev_wallet-1.addr)
        DSTADDRESS=$(cat wallet/dev_wallet-1.addr)
        CARDANO_NODE_SOCKET_PATH=/run/cardano-node-testnet.socket
        
        TRANS=$(cardano-cli query utxo $TESTNET --address $ADDRESS | tail -n1)
        UTXO=$(echo ${TRANS} | awk '{print $1}')
        ID=$(echo ${TRANS} | awk '{print $2}')
        TXIN=${UTXO}#${ID}
        
        cardano-cli transaction build \
        --testnet-magic  1097911063 \
        --alonzo-era \
        --tx-in ${TXIN} \
        --change-address ${DSTADDRESS} \
        --out-file tx.raw

2. Sign the transaction:

        cardano-cli transaction sign \
        --tx-body-file tx.raw \
        --signing-key-file wallet/dev_wallet-1.skey \
        --testnet-magic  1097911063 \
        --out-file tx.signed

3. Encode the transaction as cbor file:

    jq .cborHex tx.signed  | xxd -r -p > tx.signed.cbor

4. Submit the transaction:

        API_HOST=<hostname>
        API_PORT=80
        TXID=$(cardano-cli transaction txid  --tx-file tx.signed)
        
        curl -H "Content-Type: application/cbor" \
        --data-binary @tx.signed.cbor \
        -X POST \
        http://${API_HOST}:${API_PORT}/api/submit/tx

5. The following error message is displayed at the console:

    Something went wrong

and on the logs of the submi-api, the messages mentioned above.

**Expected behavior**
The following messages were displayed when submitting the same file with the same command after downgrading to 1.33.0:
At the console:

    "8347016ebf0f45a8cce982cf3619163f372f0324a23ac6706d3163d9b033d3b3"

In the logs:

    Feb 28 16:19:24 <hostname> bash[95607]: [cardano-tx-submit:Info:15] [2022-02-28 14:19:24.84 UTC] txSubmitPost: successfully submitted transaction 8347016ebf0f45a8

**System info (please complete the following information):**
- OS Name: Debian (server)
- OS Version 10.11 
- Node version (output of `cardano-node --version`) (on the server. same version for cardano-submit-api):

    cardano-node 1.33.0 - linux-x86_64 - ghc-8.10
    git rev 54155a28dbdb9a17e4ffe52b68f6c58af54ff83b

- CLI version  (output of `cardano-cli --version`) (on the workstation, where the transaction was created):

    cardano-cli 1.32.1 - linux-x86_64 - ghc-8.10
    git rev 4f65fb9a27aa7e3a1873ab4211e412af780a3648

Same error with cardano-cli 1.34.0-rc3 on the workstation (and with the transaction created on the server, but signed on the workstation and submitted from the workstation):

    cardano-cli 1.33.0 - linux-x86_64 - ghc-8.10
    git rev 54155a28dbdb9a17e4ffe52b68f6c58af54ff83b

**Comment by disassembler** [comment]
Cannot reproduce running both node and submit-api on tag 1.33.0-rc3:

```
node/submit-api:
git log -n 1   
Wed Feb 23 12:37:23 2022 +0100 54155a28d (HEAD, tag: 1.34.0-rc3) Revert bump to ledger and consensus versions.  [Nicholas Clarke]
run following in separate terminals:
nix run .#testnet/node
nix run .#testnet/submit-api

run CLI in terminal with older 1.32.1:
cardano-node
  homepage: https://cardano.org
  url: https://github.com/input-output-hk/cardano-node/archive/4f65fb9a27aa7e3a1873ab4211e412af780a3648.tar.gz
  owner: input-output-hk
  branch: refs/tags/1.32.1
  url_template: https://github.com/<owner>/<repo>/archive/<rev>.tar.gz
  repo: cardano-node
  type: tarball
  sha256: 00k9fqrm0gphjji23x0nc9z6bqh8bqrncgivn3mi3csacjzicrrx
  description: The core component that is used to participate in a Cardano decentralised blockchain.
  rev: 4f65fb9a27aa7e3a1873ab4211e412af780a3648

cardano-cli --version
cardano-cli 1.32.1 - linux-x86_64 - ghc-8.10
git rev 4f65fb9a27aa7e3a1873ab4211e412af780a3648

cardano-cli transaction build --testnet-magic 1097911063 --alonzo-era --tx-in 2079ae787eef5999ff63904f50a9b0f2cf77da6015086b5db75b817a8a69c0f3#0 --change-address addr_test1vrd26p8d9dlknc4fhevzudcfzuul5qm2znquytugqcw583czzqrpm --out-file tx-test.txbody
Estimated transaction fee: Lovelace 164093

cardano-cli transaction sign --tx-body-file tx-test.txbody --out-file tx-test.txsigned --signing-key-file shelley/rich-utxo.skey

xxd -r -p <<< $(jq .cborHex tx-test.txsigned) > tx.signed.cbor

curl -i --header "Content-Type: application/cbor" -X POST http://localhost:8090/api/submit/tx --data-binary @tx.signed.cbor
HTTP/1.1 202 Accepted
Transfer-Encoding: chunked
Date: Mon, 28 Feb 2022 16:02:51 GMT
Server: Warp/3.3.18
Content-Type: application/json;charset=utf-8

"892a17e27554a18148e7a4c71c6def70c3dc9873f0a56678b2ff9521a20e6bf3"%

```

**Comment by cardano-apexpool** [comment]
After doing some more tests on this relay and on another one, it turns out that **the bug manifests only when p2p is enabled.**
I was running this testnet relay with p2p enabled. After observing that the same transaction is working on my second relay, and then again on the first relay it is not working, I disabled p2p in the config file, restarted the services (cardano-node and cardano-submit-api) and the same transaction could be submitted. Obviously it manifests only when p2p is enabled on the cardano-node.

**Comment by coot** [comment]
It is a duplicate of #3568, which was fixed in input-output-hk/ouroboros-network#3606 but it is not included in `1.34.0-rc3`.  Please reopen if you can reproduce this using `master`.

**Comment by santonode** [comment]
The submitapi needs to work in the next version. Freeloaderz.io depends on it. We are a coalition of 60+ pools server thousands of transactions every day and awarded Catalyst Fund8 for expansion
